Content.Server/Doors/Systems/DoorSystem.cs

```diff
@@ -305,6 +307,30 @@ private void OnEmagged(EntityUid uid, DoorComponent door, GotEmaggedEvent args)
         }
     }
 
+    private void OnEmagFixed(EntityUid uid, DoorComponent door, GotEmagFixedEvent args)
+    {
+        if (TryComp<AirlockComponent>(uid, out var airlockComponent))
+        {
+            if (!airlockComponent.IsPowered())
+                return;
+
+            if (airlockComponent.BoltsDown && (door.State == DoorState.Open || door.State == DoorState.Opening))
+            {
+                // The door is bolted and open. Emags do this and it can be disruptive, so undo it.
+                airlockComponent.BoltsDown = false;
+                SetState(uid, DoorState.Closing, door);
+                args.Handled = true;
+            }
+            else if (door.State == DoorState.Emagging)
+            {
+                // The door was JUST emagged and is playing the zap animation.
+                // If it's allowed to finish then the door will be opened and bolted. Simply interrupt this.
+                SetState(uid, DoorState.Closing, door);
+                args.Handled = true;
+            }
```

Comment:
does this actually work? i probably just wouldn't handle this case at all tbh, can't imagine when it would even happen

Reply:
Upon emagging a closed airlock it immediately disables the ID checker, plays the zap animation and sets the door's state to "Emagging" for 0.8 seconds, after which if uninterrupted it's hardcoded to force the "Opening" state and bolt itself. So changing the state to something else other than "Emagging" meanwhile is guaranteed to prevent that.

Main reason I add it here is that a charge is consumed in fixing the ID checker even if the door is stuck in that emag animation state, so if I didn't interrupt it you'd lose a charge while still having the door bolted open with a traitor in front of you. I expect it to happen mostly during nukie rounds.